dali/kernels/slice/slice_cpu.h

```diff
@@ -275,6 +278,91 @@ void SliceKernel(OutputType *output,
   }
 }
 
+
+/**
+ * @brief Implementation of slice kernel with an execution engine.
+ */
+template <typename ExecutionEngine, typename OutputType, typename InputType, int Dims>
+void SliceKernel(ExecutionEngine &exec_engine,
+                 OutputType *output,
+                 const InputType *input,
+                 const TensorShape<Dims> &in_strides,
+                 const TensorShape<Dims> &out_strides,
+                 const TensorShape<Dims> &anchor,
+                 const TensorShape<Dims> &in_shape,
+                 const TensorShape<Dims> &out_shape,
+                 const OutputType *fill_values,
+                 int channel_dim = -1,  // negative if no channel dim or already processed
+                 int min_blk_sz = 16000,
+                 int req_nblocks = -1) {
+  // Parallelize
+  if (req_nblocks < 0)
+    req_nblocks = exec_engine.NumThreads() * 8;
+
+  int nblocks = 1;
+  std::array<int, Dims> split_factor;
+  if (req_nblocks > 1) {
+    split_factor.fill(1);
+    // Either ``req_nblocks`` blocks or fewer if remaining block sizes < min_blk_sz
+    split_shape(split_factor, out_shape, req_nblocks, min_blk_sz);
+    nblocks = volume(split_factor);
+  }
+
+  if (nblocks == 1) {
+    SliceKernel(output, input, in_strides, out_strides, anchor, in_shape, out_shape,
+                fill_values, channel_dim);
+    return;
+  }
+
+  int last_split_dim = LastSplitDim(split_factor);
+  TensorShape<Dims> start;
+  for (int d = 0; d < Dims; d++)
+    start[d] = 0;
```

Comment:
```suggestion
  start.resize(Dims);
```
Note 1: It will handle DynamicDimensions, if/when we support them and will be a no-op otherwise
Note 2: TensorShape is zero-filled by default.